tools/firmware-utils/src/tplink-safeloader.c

```diff
@@ -182,8 +182,7 @@ static struct device_info boards[] = {
 			{"device-info", 0x31400, 0x00400},
 			{"signature", 0x32000, 0x00400},
 			{"device-id", 0x33000, 0x00100},
-			{"os-image", 0x40000, 0x1c0000},
-			{"file-system", 0x200000, 0x5b0000},
+			{"firmware", 0x40000, 0x770000},
```

Comment:
@adrianschmutzler Aren't you supposed to change `first_sysupgrade_partition` to `firmware` as well?

https://github.com/openwrt/openwrt/blob/6b2d917d0c09ad6ce594b0836291335febe60a35/tools/firmware-utils/src/tplink-safeloader.c#L195

Reply:
No, I made the same error for the CPE210v3 PR; it seems to have to be the name of dynamically created partition (all other devices also do it this way):
https://github.com/openwrt/openwrt/pull/1843#discussion_r268333861

Reply:
Ok, seems like nobody has done this change so far, weird. Anyway, any reason to make `{"firmware", 0x40000, 0x770000},` shorter by 256k ? I mean `0x40000+0x1c0000+0x5b0000 = 0x7b0000` and not `0x770000` ?

Reply:
0x770000 is the _length_?! Or am I getting something terribly wrong?

0x1c0000+0x5b0000 =0x770000 ??

Reply:
> Ok, seems like nobody has done this change so far, weird. 

From what I get out of @bluelineXY 's report, using "firmware" instead of os-image will not work:
https://github.com/openwrt/openwrt/pull/1843#discussion_r268333861

Reply:
Ok, it's base + size, so `0x40000+0x770000=0x7b0000`